frontend/public/notification.js

```diff
@@ -0,0 +1,21 @@
+window.addEventListener('load', () => {
+  if (Notification.permission !== 'granted') {
+    const bell = document.createElement('div');
+    bell.innerText = '🔔'
+    bell.style.position = 'absolute';
+    bell.style.top = bell.style.right = '1em';
+    bell.style.cursor = 'pointer';
+    bell.addEventListener('click', () => {
+      Notification.requestPermission().then((result) => {
+        if (result === 'granted') {
+          bell.remove();
+          new Notification('Success!', {
```

Comment:
Is the permission that's granted here automatically remembered for use for the rest of the app installation time?

Reply:
On Chrome and Firefox it is. The choice is persisted in the browser settings. If the user chooses either option, `Notification.requestPermission()` will never show a user-interactive request again, but always immediately return the chosen value. In fact it is quite hard to go back to a state where `Notification.requestPermission()` will show a choice again. The UI to erase the choice is hidden fairly deep in the browser settings. It is much easier to toggle between `granted` and `denied` states. In desktop browsers this is generally possible through a button in the address bar.

The one exception I found is using private/incognito sessions. Choices made there are only persisted for as long as that particular session lasts, for obvious reasons.